src/embind/embind.js

```diff
@@ -642,11 +642,8 @@ var LibraryEmbind = {
             return value;
         },
         'toWireType': function(destructors, value) {
-            // todo: Here we have an opportunity for -O3 level "unsafe" optimizations: we could
-            // avoid the following if() and assume value is of proper type.
-            if (typeof value !== "number" && typeof value !== "boolean") {
-                throw new TypeError('Cannot convert "' + _embind_repr(value) + '" to ' + this.name);
-            }
+            // The VM will perform JS to Wasm value conversion, according to the spec:
+            // https://www.w3.org/TR/wasm-js-api-1/#towebassemblyvalue
```

Comment:
How about we put the old check behind "#if ASSERTIONS`?

Reply:
Thanks Sam, I put the old code back gated by ASSERTIONS.